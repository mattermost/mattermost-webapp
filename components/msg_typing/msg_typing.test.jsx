// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import MsgTyping from 'components/msg_typing/msg_typing.jsx';

describe('components/MsgTyping', () => {
    const baseProps = {
        typingUsers: [],
        channelId: 'test',
        postId: null,
    };

    test('should match snapshot, on nobody typing', () => {
        const wrapper = shallow(<MsgTyping {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, on one user typing', () => {
        const typingUsers = ['test.user'];
        const props = {...baseProps, typingUsers};

        const wrapper = shallow(<MsgTyping {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, on multiple users typing', () => {
        const typingUsers = ['test.user', 'other.test.user', 'another.user'];
        const props = {...baseProps, typingUsers};

        const wrapper = shallow(<MsgTyping {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
