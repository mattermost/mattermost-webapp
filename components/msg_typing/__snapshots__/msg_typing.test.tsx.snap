// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MsgTyping should match snapshot, on multiple users typing 1`] = `
<span
  className="msg-typing"
>
  <FormattedMessage
    defaultMessage="{users} and {last} are typing..."
    id="msg_typing.areTyping"
    values={
      Object {
        "last": "another.user",
        "users": "test.user, other.test.user",
      }
    }
  />
</span>
`;

exports[`components/MsgTyping should match snapshot, on nobody typing 1`] = `
<span
  className="msg-typing"
/>
`;

exports[`components/MsgTyping should match snapshot, on one user typing 1`] = `
<span
  className="msg-typing"
>
  <FormattedMessage
    defaultMessage="{user} is typing..."
    id="msg_typing.isTyping"
    values={
      Object {
        "user": "test.user",
      }
    }
  />
</span>
`;
